src/core/xref-db.js

```diff
@@ -73,27 +74,30 @@ async function shouldBustCache(cache) {
   const res = await fetch(url);
   if (!res.ok) return false;
   const lastUpdated = await res.text();
-  await cache.set("__LAST_VERSION_CHECK__", now);
+  localStorage.setItem(key, now.toString());
   return parseInt(lastUpdated, 10) > lastChecked;
 }
 
 /**
- * @param {Map<string, SearchResultEntry[]>} data
+ * @param {RequestEntry[]} queries
+ * @param {Map<string, SearchResultEntry[]>} results
  */
-export async function cacheXrefData(data) {
+export async function cacheXrefData(queries, results) {
   try {
     const cache = await getIdbCache();
-    // add data to cache
-    await cache.addMany(data);
+    const tx = cache.transaction(STORE_NAME, "readwrite");
+    for (const query of queries) {
+      tx.objectStore(STORE_NAME).add({ query, result: results.get(query.id) });
+    }
```

Comment:
Bit easier on the eye... 

```suggestion
    for (const query of queries) {
      const result = results.get(query.id);
      tx.objectStore(STORE_NAME).add({ query, result });
    }
```